src/util/file-filter.js

```diff
@@ -7,24 +7,22 @@ import {createLogger} from './logger';
 
 const log = createLogger(__filename);
 
-// Use this function to mimic path.resolve without resolving to absolute path.
-export const normalizeResolve = (file: string): string => {
-  // normalize
-  file = path.normalize(file);
-
-  // trim trailing slash
-  if (path.parse(file).base && file.endsWith(path.sep)) {
-    return file.slice(0, -1);
+// check if target is a sub directory of src
+export const isSubDir = (src: string, target: string): boolean => {
+  const relate = path.relative(src, target);
+  // same dir
+  if (!relate) {
+    return false;
   }
-  return file;
+  return relate[0] !== '.';
```

Comment:
@kumar303 @eight04 If I'm reading it correctly, this check assumes that the `relate` path produced by `path.relative` is going to contain a `"."` as its first character only if the path is not a subdir (e.g. "../artifact"), which can be a wrong assumption if the file name starts with a `"."` (e.g. "web-ext build -s . -a .artifact")

In that case the artifact dir ".artifact" is not going to be recognized as a subdir anymore, but it is currently excluded by the default filter list (which contains "**/.*"), nevertheless if we are going to allow the customization of the base ignored patterns, the artifact dir content can be included by mistake in the zip file.

How about making this check more explicit? e.g. `return !relate.startsWith("../");`

Nit, This helper do not check if target is a dir, to make it more clear we could name it `isSubPath`.